### ISSUE TITLE
Add python-dateutil to deps

### ISSUE THREAD

**Issue by nvn-nil** [octue/octue-sdk-python#179]
### What is the current behavior?

*If the current behavior is a bug, please provide any stack traces the steps reproduce and, if possible, a minimal demo of the problem.*

I'm using octue to ask a question and in my script the import I need are 
```
import json
import os
from octue.cloud.pub_sub.service import Service
from octue.resources import Datafile, Dataset, Manifest
from octue.resources.service_backends import GCPPubSubBackend
```

So I did 
`poetry add octue=0.1.19` and while running my script I get hit with
```
from octue.utils.decoders import OctueJSONDecoder
  File "C:\Users\naveen\AppData\Local\pypoetry\Cache\virtualenvs\tesk-hL0eLDuY-py3.9\lib\site-packages\octue\utils\decoders.py", line 3, in <module>
    import dateutil.parser
ModuleNotFoundError: No module named 'dateutil'
```

So we need to add `python-dateutil` to the dependencies

### What is the expected behavior?

I can ask a question using only octue and builtin libraries

### Your environment

  - Library Version: `0.1.19`
  - Platform Windows and WSL
  - Python 3.9.4

**Comment by cortadocodes** [comment]
Thanks for raising this @nvn-nil! It's odd that `dateutil` isn't needed as an explicit dependency locally or on our test runners for the tests to pass (I checked to make sure that the `dateutil` usage is covered). Anyway, I'll fix this very soon 👍

**Comment by cortadocodes** [comment]
@nvn-nil do you need this fix urgently or am I ok to put it out in the next release (probably next week)?

**Comment by thclark** [comment]
Its not urgent @cortadocodes

**Comment by cortadocodes** [comment]
@nvn-nil I forgot to ask what version of python did you experience this bug on?

**Comment by nvn-nil** [comment]
> @nvn-nil I forgot to ask what version of python you experienced this bug on?

Python 3.9.4 64bit in windows

**Comment by cortadocodes** [comment]
Fixed in [0.1.22](https://github.com/octue/octue-sdk-python/releases/tag/0.1.22)
